server/src/domain/media/media.service.ts

```diff
@@ -49,6 +52,32 @@ export class MediaService {
       }
     }
 
+    const people = force ? await this.personRepository.getAll() : await this.personRepository.getAllWithoutThumbnail();
+
+    for (const person of people) {
+      // use stored asset for generating thumbnail or pick a random one if not present
+      const face = person.faceAssetId
+        ? await this.personRepository.getFaceById({ personId: person.id, assetId: person.faceAssetId })
+        : await this.personRepository.getRandomFace(person.id);
+      if (face) {
+        await this.jobRepository.queue({
+          name: JobName.GENERATE_FACE_THUMBNAIL,
+          data: {
+            imageWidth: face.imageWidth,
+            imageHeight: face.imageHeight,
+            boundingBox: {
+              x1: face.boundingBoxX1,
+              x2: face.boundingBoxX2,
+              y1: face.boundingBoxY1,
+              y2: face.boundingBoxY2,
+            },
+            assetId: face.assetId,
+            personId: person.id,
+          } as IFaceThumbnailJob,
```

Comment:
this one as well

Reply:
You're right! I just saw that every combination of queue and parameters are typed.